integration-tests/tests/network/runner.rs

```diff
@@ -408,14 +412,19 @@ impl StateMachine {
                                                     })
                                                 },
                                             );
-
+                                        debug!(target: "network", "{}: ping, pong check : {} {:?} {:?} expected {:?} {:?}",
+                                            num_prev_actions, source, pings, pongs, pings_expected, pongs_expected);
                                         if ping_ok && pong_ok {
                                             flag.store(true, Ordering::Relaxed);
                                         } else {
-                                            panic!(
-                                                "ping, pong check failed got: {:?} {:?}",
-                                                pings, pongs
-                                            );
+                                            if time.lock().unwrap().elapsed()
```

Comment:
I’d prefer `} else if {` to reduce amount of indentation.

Reply:
@mina86 I removed that else branch completely.